File: MySolutions/Python/q189_rotateArray.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l=len(nums)
        k = k%l

        nums[:] = nums[l-k:]+nums[:l-k]  #rotate array right by k places
        
        #nums[:] = nums[k:]+nums[:k]    rotate array left by k places
        

File: MySolutions/Cpp/q189_rotateArray.cpp
//time wise - beats 100%

class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int l = nums.size();
        k = k % l;
        
        // Perform rotation in-place
        vector<int> temp(nums.begin() + l - k, nums.end());
        nums.erase(nums.begin() + l - k, nums.end());
        nums.insert(nums.begin(), temp.begin(), temp.end());
    }
};